BackEnd/src/business/stocks.ts

```diff
@@ -0,0 +1,156 @@
+import TeamModel from '../../../Common/models/team.model';
+import { GetDbTeamRosterByTeamId, GetDbTeamsByTeamId } from '../db/teams';
+import { InvalidRequestError } from '../../../Common/errors';
+import { AuthUser } from './auth';
+import { buyDbStockForUser, createDBNewStockValue, getDbLatestStockValue, withdrawDbFunds } from '../db/stocks';
+import { GetRiotLeagueBySummonerId, GetRiotPlayerByPuuid } from '../external-api/player';
+import { RiotLeagueEntryDto } from '../../../Common/Interface/RiotAPI/RiotApiDto';
+import PlayerGameModel from '../../../Common/models/playergame.model';
+import { GetDbPlayerGamesByGameId } from '../db/games';
+import { ToGameId } from '../../../Common/utils';
+import { getDbPlayerTeamPlayerPuuid } from '../db/playerteam';
+import logger from '../../logger';
+
+export async function updateTeamStocksForGame(matchId: string): Promise<void> {
+  logger.info(`Updating Stock Values For Match: ${matchId}`);
+  const allPlayersGames: PlayerGameModel[] = await GetDbPlayerGamesByGameId(ToGameId(matchId));
+
+  // We need to do this because sometimes players arent registered to a team. (Eg, ESUBs)
+  let loserTeamIds: Map<number, number> = new Map<number, number>();
+  let winningTeamIds: Map<number, number> = new Map<number, number>();
+  let seasonIds: Map<number, number> = new Map<number, number>();
+
+  for (let playerGame of allPlayersGames) {
+    const teamId: number = await getDbPlayerTeamPlayerPuuid(playerGame.playerPuuid, playerGame.seasonId);
+    seasonIds.set(playerGame.seasonId, (seasonIds.get(playerGame.seasonId) || 0) + 1);
+
+    if(teamId == null) {
+      continue;
+    }
+
+    if(playerGame.win) {
+      winningTeamIds.set(teamId, (winningTeamIds.get(teamId) || 0) + 1);
+    } else {
+      loserTeamIds.set(teamId, (loserTeamIds.get(teamId) || 0) + 1);
+    }
+  }
+
+  if(winningTeamIds.size == 0 || loserTeamIds.size == 0 || seasonIds.size == 0) {
+    logger.info(`There was a missing winningIds or missing losingIds for match ${matchId}`);
+    return;
+  }
+
+  let winningTeamId = [...winningTeamIds.entries()].reduce((a, e ) => e[1] > a[1] ? e : a)[0];
+  let losingTeamId = [...loserTeamIds.entries()].reduce((a, e ) => e[1] > a[1] ? e : a)[0];
```

Comment:
ahhhhh e and a ahhhhh

Reply:
Oh woops, grabbed this code off stackoverflow LOL. Ill update in the next PR